cg/utils/utils.py

```diff
@@ -20,3 +20,10 @@ def build_command_from_dict(options: dict, exclude_true: bool = False) -> List[s
                 formatted_options.append(str(option_name))
                 formatted_options.append(str(option_value))
     return formatted_options
+
+
+def get_hamming_distance(str_1: str, str_2: str) -> int:
+    """Return the hamming distance between two strings."""
+    if len(str_1) != len(str_2):
+        raise KeyError("The two strings must have the same length to calculate distance!")
```

Comment:
I think it is good if we start documenting which exceptions a method can raise in the doc string. Here is an example:
```python
def fetch_smalltable_rows(
    table_handle: smalltable.Table,
    keys: Sequence[bytes | str],
    require_all_keys: bool = False,
) -> Mapping[bytes, tuple[str, ...]]:
    """Fetches rows from a Smalltable.

    Retrieves rows pertaining to the given keys from the Table instance
    represented by table_handle.  String keys will be UTF-8 encoded.

    Args:
        table_handle: An open smalltable.Table instance.
        keys: A sequence of strings representing the key of each table
          row to fetch.  String keys will be UTF-8 encoded.
        require_all_keys: If True only rows with values set for all keys will be
          returned.

    Returns:
        A dict mapping keys to the corresponding table row data
        fetched. Each row is represented as a tuple of strings. For
        example:

        {b'Serak': ('Rigel VII', 'Preparer'),
         b'Zim': ('Irk', 'Invader'),
         b'Lrrr': ('Omicron Persei 8', 'Emperor')}

        Returned keys are always bytes.  If a key from the keys argument is
        missing from the dictionary, then that row was not found in the
        table (and require_all_keys must have been False).

    Raises:
        IOError: An error occurred accessing the smalltable.
    """
```

Reply:
We have not settled on a standard for documenting our functions, so feel free to disregard this comment. But consider at least documenting the exceptions